yeonjae/programmers-1/시저암호.js

```diff
@@ -0,0 +1,18 @@
+function solution(s, n) {
+  let answer = "";
+
+  answer = s
+    .split("")
+    .map((item) => {
+      if (item === " ") {
+        return " ";
+      }
+      const charCode = item.charCodeAt();
+      return item.toUpperCase().charCodeAt() + n > 90
+        ? String.fromCharCode(charCode + n - 26)
+        : String.fromCharCode(charCode + n);
+    })
+    .join("");
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(s, n) {
  const answer = s
    .split("")
    .map((item) => {
      const nextCharCode = item.charCodeAt() + n;
        
      if (item === " ") return " ";
    
      return item.toUpperCase().charCodeAt() + n > 90
        ? String.fromCharCode(nextCharCode - 26)
        : String.fromCharCode(nextCharCode);
    })
    .join("");
  return answer;
}
```
- `toUpperCase`를 해줘서 오버플로우 했는지 조건식을 하나(대문자인 경우)로만 가져가는게 좋은 아이디어였네요 👍
- `let answer` 선언 후에 할당하기보다 바로 `const`로 선언하는게 좋아보입니다
  - 아니면 애초에 `return s.map().join()` 도 괜찮아보입니다
- 그리고 변수 선언은 사전 작업이 필요한 특수한 경우가 아니면 `최상단`으로 올려주세요 ~
- `charCode + n`이 반복되고 있는데 charCode는 딱히 필요없어보여서 `nextCharCode`를 활용하게끔 수정해봤습니다